pyserini/util.py

```diff
@@ -103,3 +133,55 @@ def download_and_unpack_index(url, index_directory='indexes', force=False, verbo
     tarball.extractall(index_directory)
     tarball.close()
     os.remove(local_tarball)
+
+def download_to_cache_and_unpack_index(url, md5, force=False, verbose=True):
+    index_name = url.split('/')[-1]
+    index_name = re.sub('''.tar.gz.*$''', '', index_name)
+    gaggle_cache_home = get_cache_home()
+    cache_index_directory = os.path.join(gaggle_cache_home, 'indexes')
+    if not os.path.exists(cache_index_directory):
+        if verbose:
+            print(f'{cache_index_directory} is created!')
+        os.makedirs(cache_index_directory)
+    local_tarball = os.path.join(cache_index_directory, f'{index_name}.tar.gz')
+    if verbose:
+        print(f'Downloading index at {url}...')
+
+    # Check to see if index already exists, if so, simply return (quietly) unless force=True, in which case we remove
+    # index and download fresh copy.
+    prebuilt_index_dir = os.path.join(cache_index_directory, f'{index_name}{md5}')
+    if os.path.exists(prebuilt_index_dir):
+        if verbose:
+            print(f'{prebuilt_index_dir} already exists!')
+        if not force:
+            if verbose:
+                print(f'Skipping download.')
+            return prebuilt_index_dir
+        if verbose:
+            print(f'force=True, removing {prebuilt_index_dir}; fetching fresh copy...')
+        shutil.rmtree(prebuilt_index_dir)
+
+    download_url(url, cache_index_directory, verbose=False, md5=md5)
+    index_path = os.path.join(cache_index_directory, index_name)
+    destination_path = index_path + compute_md5(local_tarball)
+    if verbose:
+        print(f'Extracting {local_tarball} into {destination_path}...')
+    tarball = tarfile.open(local_tarball)
+    tarball.extractall(cache_index_directory)
+    tarball.close()
+    os.remove(local_tarball)
+    os.rename(index_path, destination_path)
+    return destination_path
+
+
+def download_index(index_name, force=False, verbose=True, mirror=None):
```

Comment:
Can we make sure `prebuilt` is in the name to make it clear what the method does?